tests/integration/targets/ipam_range/tasks/main.yml

```diff
@@ -0,0 +1,283 @@
+        start: "{{ range_start }}"
+        end: "{{ range_end }}"
+        space: "{{ _ip_space.id }}"
+        state: "present"
+        name: "Test Range"
+      register: range
+    - name: Get information about the Range
+      infoblox.bloxone.ipam_range_info:
+        filters:
+          start: "{{ range_start }}"
+          end: "{{ range_end }}"
+          space: "{{ _ip_space.id }}"
+      register: range_info
+    - assert:
+        that:
+          - range is not failed
+          - range_info.objects | length == 1
+          - range_info.objects[0].name == "Test Range"
+
+    # Test case for creating a range with exclusion ranges
+    - name: "Create a Range with exclusion ranges"
+      infoblox.bloxone.ipam_range:
+        start: "{{ range_start }}"
+        end: "{{ range_end }}"
+        space: "{{ _ip_space.id }}"
+        state: "present"
+        exclusion_ranges:
+          - start: "10.0.0.10"
+            end: "10.0.0.20"
+            comment: "Exclude this range"
+      register: range
+    - name: Get information about the Range
+      infoblox.bloxone.ipam_range_info:
+        filters:
+          start: "{{ range_start }}"
+          end: "{{ range_end }}"
+          space: "{{ _ip_space.id }}"
+      register: range_info
+    - assert:
+        that:
+          - range is not failed
+          - range_info.objects | length == 1
+          - range_info.objects[0].exclusion_ranges[0].start == "10.0.0.10"
+          - range_info.objects[0].exclusion_ranges[0].end == "10.0.0.20"
+          - range_info.objects[0].exclusion_ranges[0].comment == "Exclude this range"
+
+    # Test case for creating a range with a threshold
+    - name: "Create a Range with threshold"
+      infoblox.bloxone.ipam_range:
+        start: "{{ range_start }}"
+        end: "{{ range_end }}"
+        space: "{{ _ip_space.id }}"
+        state: "present"
+        threshold:
+          enabled: true
+          high: 90
+          low: 10
+      register: range
+    - name: Get information about the Range
+      infoblox.bloxone.ipam_range_info:
+        filters:
+          start: "{{ range_start }}"
+          end: "{{ range_end }}"
+          space: "{{ _ip_space.id }}"
+      register: range_info
+    - assert:
+        that:
+          - range is not failed
+          - range_info.objects | length == 1
+          - range_info.objects[0].threshold.enabled == true
+          - range_info.objects[0].threshold.high == 90
+          - range_info.objects[0].threshold.low == 10
+
+  always:
+    # Cleanup if the test fails
+    - name: "Delete a Range"
+      infoblox.bloxone.ipam_range:
+        start: "{{ range_start }}"
+        end: "{{ range_end }}"
+        space: "{{ _ip_space.id }}"
+        state: "absent"
+
+    - name: "Delete Subnet"
+      infoblox.bloxone.ipam_subnet:
+        address: "10.0.0.0/24"
+        space: "{{ _ip_space.id }}"
+        state: "absent"
+      ignore_errors: true
+
+    - name: "Delete IP Space"
```

Comment:
Can we have a standardized cleanup file in `setup_ip_space` instead of this and call that from here ?
Something like :

```yaml
- ansible.builtin.include_role:
    name: "setup_ip_space"
    tasks_from: "cleanup.yml"

Reply:
This isnt resolved yet.